homeassistant/components/nest/config_flow.py

```diff
@@ -240,72 +249,77 @@ async def async_step_device_project(
             errors=errors,
         )
 
-    async def async_step_pubsub(
+    async def async_step_pubsub_topic(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
-        """Configure and the pre-requisites to configure Pub/Sub topics and subscriptions."""
-        data = {
-            **self._data,
-            **(user_input if user_input is not None else {}),
-        }
-        cloud_project_id = data.get(CONF_CLOUD_PROJECT_ID, "").strip()
```

Comment:
why did we strip in the past?

Reply:
There is a test in `test_pubsub_subscription_strip_whitespace` but it looks like it doesn't work.

The reason this was here is that users would copy and paste these values often add in extra spaces, then get weird errors. I'm going to go ahead and add this back just to be nice to users.

I removed the test since it doesn't work anyway.